Neos.Flow/Classes/SignalSlot/Dispatcher.php

```diff
@@ -82,11 +87,25 @@ public function connect($signalClassName, $signalName, $slotClassNameOrObject, $
             $method = $slotMethodName;
         }
 
+        if (is_object($slotClassNameOrObject) || class_exists($slotClassNameOrObject)) {
+            try {
+                if ($slotClassNameOrObject instanceof \Closure) {
+                    $methodParameters = (new \ReflectionFunction($slotClassNameOrObject))->getParameters();
+                } else {
+                    $methodParameters = (new \ReflectionMethod($slotClassNameOrObject, trim($method, ':')))->getParameters();
+                }
```

Comment:
Nice trick to make the change backwards compatible! But is it OK to use reflection during runtime? Or – if not – could this be put into a `compile static` function?

Reply:
Turned out `class_exists()` broke everything during compile phase. So, bad idea either way. The ReflectionService cannot be used here, as it is not (always) available at this point.